internal/provider/resource_port_forward.go

```diff
@@ -0,0 +1,180 @@
+package provider
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/paultyng/go-unifi/unifi"
+)
+
+//	GetPortForward(site string, id string) (*unifi.PortForward, error)
+//	DeletePortForward(site string, id string) (*unifi.PortForward, error)
+//	CreatePortForward(site string, d *unifi.PortForward) (*unifi.PortForward, error)
+//	UpdatePortForward(site string, d *unifi.PortForward) (*unifi.PortForward, error)
+
+func resourcePortForward() *schema.Resource {
+	return &schema.Resource{
+		Create: resourcePortForwardCreate,
+		Read:   resourcePortForwardRead,
+		Update: resourcePortForwardUpdate,
+		Delete: resourcePortForwardDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"dst_port": {
+				Type:         schema.TypeString,
+				Required:     false,
+				Optional:     true,
+				ValidateFunc: validation.StringMatch(portForwardDstPortRegexp, "dst_port is invalid"),
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"fwd": {
+				Type:         schema.TypeString,
+				Required:     false,
+				Optional:     true,
+				ValidateFunc: validation.StringMatch(portForwardFwdRegexp, "fwd is invalid"),
+			},
+			"fwd_port": {
+				Type:         schema.TypeString,
+				Required:     false,
+				Optional:     true,
+				ValidateFunc: validation.StringMatch(portForwardFwdPortRegexp, "fwd_port is invalid"),
+			},
+			"log": {
+				Type:     schema.TypeBool,
+				Default:  false,
+				Optional: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: false,
+				Optional: true,
+			},
+			"port_forward_interface": {
+				Type:         schema.TypeString,
+				Required:     false,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"wan", "wan2", "both", ""}, false),
```

Comment:
You don't need to include an empty option if this is optional I believe:
```suggestion
				ValidateFunc: validation.StringInSlice([]string{"wan", "wan2", "both"}, false),
```

Reply:
yes, I was planning to change this (but was late last night and quickly pushed the WIP PR) , as it would  be stupid if you were coding the port_forward interface if I was working on it as well. Didn't have a clear way of notifying you, so I thought creating this WIP PR was good to get the discussion going